mithril-aggregator/src/runtime/runner.rs

```diff
@@ -1236,4 +1285,94 @@ pub mod tests {
         assert!(!signed_entities.is_empty());
         assert!(!signed_entities.contains(&SignedEntityTypeDiscriminants::CardanoTransactions));
     }
+
+    #[tokio::test]
+    async fn is_outdated_return_false_when_message_is_not_expired_and_has_same_signed_entity_type()
+    {
+        assert!(!is_outdated_returned_when(IsExpired::No, true).await);
+    }
+
+    #[tokio::test]
+    async fn is_outdated_return_true_when_message_is_expired_and_has_same_signed_entity_type() {
+        assert!(is_outdated_returned_when(IsExpired::Yes, true).await);
+    }
+
+    #[tokio::test]
+    async fn is_outdated_return_true_when_message_is_not_expired_and_has_not_the_same_signed_entity_type(
+    ) {
+        assert!(is_outdated_returned_when(IsExpired::No, false).await);
+    }
+
+    #[tokio::test]
+    async fn is_outdated_return_true_when_message_is_expired_and_has_not_the_same_signed_entity_type(
+    ) {
+        assert!(is_outdated_returned_when(IsExpired::Yes, false).await);
+    }
+
+    // The network field in the signed entity configuration is modified to test the scenario
+    // where the computed signed entity type differs from the one specified in the open message.
+    // This is the simplest attribute to manipulate, compared to the protocol parameters,
+    // even though it may not be a real use case.
```

Comment:
This manipulation of the network is also confusing. Can we do another modification of the signed entity type to be closer to the reality and remove this comment?

Reply:
Done ✅